pkg/manager/member/tidb_member_manager.go

```diff
@@ -297,6 +297,110 @@ func (tmm *tidbMemberManager) syncTiDBClientCerts(tc *v1alpha1.TidbCluster) erro
 	return tmm.certControl.Create(controller.GetOwnerRef(tc), certOpts)
 }
 
+func (tmm *tidbMemberManager) syncTiDBService(tc *v1alpha1.TidbCluster) error {
+
+	newSvc := getNewTiDBServiceOrNil(tc)
+	// TODO: delete tidb service if user remove the service spec deliberately
+	if newSvc == nil {
+		return nil
+	}
+
+	ns := newSvc.Namespace
+
+	oldSvcTmp, err := tmm.svcLister.Services(ns).Get(newSvc.Name)
+	if errors.IsNotFound(err) {
+		err = SetServiceLastAppliedConfigAnnotation(newSvc)
+		if err != nil {
+			return err
+		}
+		return tmm.svcControl.CreateService(tc, newSvc)
+	}
+	if err != nil {
+		return err
+	}
+	oldSvc := oldSvcTmp.DeepCopy()
+
+	// Adopt orphaned service by simply overriding
+	if metav1.GetControllerOf(oldSvc) == nil {
```

Comment:
Adopting orphaned objects is a systematic problem that should be considered at project level in a separate issue, this PR adopts the TiDB service only.

Reply:
https://github.com/pingcap/tidb-operator/issues/1246